chain-time/src/timeframe.rs

```diff
@@ -0,0 +1,168 @@
+use crate::timeline::Timeline;
+use std::time::{Duration, SystemTime};
+
+/// Identify a slot in a *specific* timeframe
+///
+/// The slots are not comparable to others slots made on a
+/// different time frame
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct Slot(pub(crate) u64);
+
+impl From<Slot> for u64 {
+    fn from(s: Slot) -> u64 {
+        s.0
+    }
+}
+
+/// Identify a slot in a specific timeframe and a leftover duration
+#[derive(Debug)]
+pub struct SlotAndDuration {
+    pub slot: Slot,
+    /// The offset of a specific time frame in
+    pub slot_duration_offset: Duration,
```

Comment:
Just `offset` (as a reader method) would suffice.

Reply:
The type of this might want to be `SlotDuration` for extra checking, but maybe convenience suggests otherwise.

Reply:
I've changed the name, but no I spefically want the extra precision from Duration (think nanosecs) here